scripts/cleanup/unusued_compilation.py

```diff
@@ -0,0 +1,129 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+import argparse
+import fnmatch
+import os
+import shutil
+import sys
+sys.path.append(os.path.join(os.path.dirname(os.path.abspath(__file__)), '../build'))
+import ci_build
+
+EXTENSIONS_OF_INTEREST = ('.c', '.cc', '.cpp', '.cxx', '.h', '.hpp', '.hxx', '.inl')
+EXCLUSIONS = (
+    'AZ_DECLARE_MODULE_CLASS(', 
+    'REGISTER_QT_CLASS_DESC(',
+    'TEST(',
+    'TEST_F(',
+    'INSTANTIATE_TEST_CASE_P(',
+    'INSTANTIATE_TYPED_TEST_CASE_P(',
+    'AZ_UNIT_TEST_HOOK(',
+    'IMPLEMENT_TEST_EXECUTABLE_MAIN(',
+    'DllMain(',
+    'CreatePluginInstance('
+)
+PATH_EXCLUSIONS = (
+    '*\\Platform\\Android\\*',
+    '*\\Platform\\Common\\*',
+    '*\\Platform\\iOS\\*',
+    '*\\Platform\\Linux\\*',
+    '*\\Platform\\Mac\\*',
```

Comment:
I guess it's possible that this script could flag code that is only used on a platform other than Windows, but considering it just gives a list of things that are potential candidates for deletion that's completely fine, and one could argue that any such code should be located somewhere inside the Platform folder anyway.

Reply:
yeah, I did find a case where we had platform-specific code outside of the `Platform` folder (AzCore/Android), so I had to exclude it explicitly.
And yes, this produces a lot of false-positives, files that are platform-specific but outside `Platform` folder is the most common case. 
For platform specific code I think we should do the inverse, analyzing just that code, that will accelerate the search.
I honestly think is not worthy to scan that, most of the unused code is in legacy pieces (Editor, CryCommon, CrySystem, Maestro, LmbrCentral, etc)